tests/test_directives.py

```diff
@@ -19,6 +19,15 @@ def available_directives_list():
             PROCESS_FRACTION]
 
 
+@pytest.fixture()
+def directives_fixture(available_directives_list):
+    def _directives(idx=None):
+        return Directives(available_directives_list) if idx is None \
+            else Directives(available_directives_list[:idx])
+
+    return _directives
+
+
```

Comment:
There really is no need for the internal function. We can just return a `Directives` object.

Reply:
I think returning `Directives(available_directives_list)` will be a good choice here. But since in some methods we're using more than one instance of `Directives` class, I think this fixture is also good due to the pass by reference behavior of python class.

Reply:
In the mentioned case, I now have initialized the instance manually.